### PR TITLE
Flower

### PR DESCRIPTION
**Pull request by EliasGuedra**

### CODE REVIEW

File: tutorials/flower/mnist-pytorch/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## Setup the Server (The virual machine)
+First connect to the VM via SSH and then run the following commands in any folder you would like.
+````bash
+git clone https://github.com/aidotse/EdgeLab.git
+cd EdgeLab/tutorials/flower/mnist-pytorch/server
+````
+To start the serverdocker now run these comands:
+````bash
+sudo docker build --tag server-docker .
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-pytorch/README.md]: Change the image name from server-docker to flower_pytorch_server.

File: tutorials/flower/mnist-pytorch/client/build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+IMAGE_NAME="client-docker"
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-pytorch/client/build.sh]: Change the image name from client-docker to flower_pytorch_client.

File: tutorials/flower/mnist-pytorch/client/client.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
+from collections import OrderedDict
+import warnings
+
+import flwr as fl
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.transforms as transforms
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10, MNIST
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-pytorch/client/client.py]: Remove CIFAR10.

File: tutorials/flower/mnist-pytorch/client/client.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            _, predicted = torch.max(outputs.data, 1)
+            total += labels.size(0)
+            correct += (predicted == labels).sum().item()
+    loss /= len(testloader.dataset)
+    accuracy = correct / total
+    return loss, accuracy
+
+
+def load_data():
+    """Load CIFAR-10 (training and test set)."""
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-pytorch/client/client.py]: CIFAR-10 -> MNIST

File: tutorials/flower/mnist-pytorch/client/client.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+# #############################################################################
+
+
+def main():
+    """Create model, load data, define Flower client, start Flower client."""
+
+    # Load model
+    net = Net().to(DEVICE)
+
+    # Load data (CIFAR-10)
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-pytorch/client/client.py]: CIFAR-10 -> MNIST

File: tutorials/flower/mnist-pytorch/client/run.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+IMAGE_NAME="client-docker"
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-pytorch/client/run.sh]: Change the image name from client-docker to flower_pytorch_client.

File: tutorials/flower/mnist-tensorflow/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+## Setup the Server (The virual machine)
+First connect to the VM via SSH and then run the following commands in any folder you would like.
+````bash
+git clone https://github.com/aidotse/EdgeLab.git
+cd EdgeLab/tutorials/flower/mnist-tensorflow/server
+````
+To start the serverdocker now run these comands:
+````bash
+sudo docker build --tag server-docker .
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-tensorflow/README.md]: Change the image name from server-docker to flower_pytorch_server.

File: tutorials/flower/mnist-tensorflow/client/build.sh [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+IMAGE_NAME="client-docker"
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-tensorflow/client/build.sh]: Change the image name from client-docker to flower_pytorch_client.

File: tutorials/flower/mnist-tensorflow/client/client.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    # Instanstiate client
+    client = MnistClient()
+
+    # Start client
+    print("Connecting to " + SERVER_IP)
+    fl.client.start_numpy_client(server_address=SERVER_IP, client=client)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-tensorflow/client/client.py]: New line.

File: tutorials/flower/mnist-tensorflow/client/run.sh [hunk 10]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+IMAGE_NAME="client-docker"
+
+ROOT_DIR=$PWD
+
+docker run -it --rm --runtime nvidia -e SERVER_IP=$1 \
+-v $ROOT_DIR:/app \
+$IMAGE_NAME
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-tensorflow/client/run.sh]: New line.

File: tutorials/flower/mnist-tensorflow/server/server.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        min_fit_clients=2,
+        min_available_clients=2,
+    )
+
+    # Start server
+    fl.server.start_server(
+        config={"num_rounds": 6},
+        strategy=strategy,
+        force_final_distributed_eval = True,
+    )
```
Comment (eriks-aidotse) [inline tutorials/flower/mnist-tensorflow/server/server.py]: New line.

**Review by eriks-aidotse — COMMENTED** [review]
The README.md files in flower/mnist-pytorch and flower/mnist-tensorflow are more or less the same. I suggest that you put everything that is the same README.md file one level up, that is, in flower/README.md. Perhaps the flower/mnist-pytorch/README.md and flower/mnist-tensorflow/README.md will be redundant.
